virtual-desktop/src/app/plugin-manager/shared/plugin-manager.ts

```diff
@@ -25,20 +25,26 @@ export class PluginManager implements MVDHosting.PluginManagerInterface {
     this.loadApplicationPluginDefinitionsMap();
   }
 
+  clearApplicationPluginDefinitions(): void {
+    this._pluginDefinitions.clear();
+  }
+
   loadApplicationPluginDefinitions(): Promise<MVDHosting.DesktopPluginDefinition[]> {
     if (this._pluginDefinitions != null) {
-      return Promise.resolve(Array.from(this._pluginDefinitions.values()));
+      if (this._pluginDefinitions.size != 0) {
```

Comment:
Just make this a single if with && unless there was some other condition to deal with here